### PR TITLE
Feat: New RSS package!

### PR DESCRIPTION
**Pull request by bholmesdev**
This documents our shiny new `@astrojs/rss` package (coming soon to an `npm` near you 👀 ). Some notable additions:
- **New "Using `@astrojs/rss`"** on our RSS guide
- **Revisions to the old `getStaticPaths` docs.** This will _not_ be removed for Astro 1.0, so we'll maintain this documentation for legacy purposes. 
- **A new "default environment variables" section** on our env variable docs. We introduced a new `import.meta.env.SITE` variable to pull off RSS, so I felt this deserved some documentation.

### CODE REVIEW

File: src/pages/en/guides/environment-variables.md [hunk 1]
```diff
@@ -17,7 +17,20 @@ SECRET_PASSWORD=password123
 PUBLIC_ANYBODY=there
 ```
 
-In this example, `PUBLIC_ANYBODY` will be available in server or client code, while `SECRET_PASSWORD` will not.
+In this example, `PUBLIC_ANYBODY` (accessible via `import.meta.env.PUBIC_ANYBODY`) will be available in server or client code, while `SECRET_PASSWORD` (accessible via `import.meta.env.SECRET_PASSWORD`) will be server-side only.
```
Comment (sarah11918) [inline src/pages/en/guides/environment-variables.md:20]:
```suggestion
In this example, `PUBLIC_ANYBODY` (accessible via `<ImportMetaEnv path=".PUBLIC_ANYBODY" />`) will be available in server or client code, while `SECRET_PASSWORD` (accessible via `<ImportMetaEnv path=".SECRET_PASSWORD" />`) will be server-side only.
```
We'll need to use the fancy new component to render this expression properly. :nail_care:

File: src/pages/en/guides/environment-variables.md [hunk 2]
```diff
@@ -17,7 +17,20 @@ SECRET_PASSWORD=password123
...
+
+## Default environment Variables
+
+Astro includes a few environment variables out-of-the-box:
+
+- `import.meta.env.MODE ('development' | 'production')`: the mode your site is running in. This is `development` when running `astro dev` and `production` when running `astro build`.
```
Comment (sarah11918) [inline src/pages/en/guides/environment-variables.md:26]:
```suggestion
- `<ImportMetaEnv path=".MODE" /> ('development' | 'production')`: the mode your site is running in. This is `development` when running `astro dev` and `production` when running `astro build`.
```

File: src/pages/en/guides/environment-variables.md [hunk 3]
```diff
@@ -17,7 +17,20 @@ SECRET_PASSWORD=password123
...
+
+- `import.meta.env.BASE_URL (string)`: the base url your site is being served from. This is determined by the [base config option](/en/reference/configuration-reference/#base).
```
Comment (sarah11918) [inline src/pages/en/guides/environment-variables.md:28]:
```suggestion
- `<ImportMetaEnv path=".BASE_URL" /> (string)`: the base url your site is being served from. This is determined by the [base config option](/en/reference/configuration-reference/#base).
```

File: src/pages/en/guides/environment-variables.md [hunk 4]
```diff
@@ -17,7 +17,20 @@ SECRET_PASSWORD=password123
...
+
+- `import.meta.env.PROD (boolean)`: whether your site is running in production.
```
Comment (sarah11918) [inline src/pages/en/guides/environment-variables.md:30]:
```suggestion
- `<ImportMetaEnv path=".PROD" /> (boolean)`: whether your site is running in production.
```

File: src/pages/en/guides/environment-variables.md [hunk 5]
```diff
@@ -17,7 +17,20 @@ SECRET_PASSWORD=password123
...
+
+- `import.meta.env.DEV (boolean)`: whether your site is running in development (always the opposite of `import.meta.env.PROD`).
```
Comment (sarah11918) [inline src/pages/en/guides/environment-variables.md:32]:
```suggestion
- `<ImportMetaEnv path=".DEV" /> (boolean)`: whether your site is running in development (always the opposite of `import.meta.env.PROD`).
```

File: src/pages/en/guides/environment-variables.md [hunk 6]
```diff
@@ -17,7 +17,20 @@ SECRET_PASSWORD=password123
...
+- `import.meta.env.SITE (string)`: [The `site` option](/en/reference/configuration-reference/#site) specified in your project's `astro.config`.
```
Comment (sarah11918) [inline src/pages/en/guides/environment-variables.md:33]:
```suggestion
- `<ImportMetaEnv path=".SITE" /> (string)`: [The `site` option](/en/reference/configuration-reference/#site) specified in your project's `astro.config`.
```

File: src/pages/en/guides/rss.md [hunk 7]
```diff
@@ -6,11 +6,139 @@ description: An intro to RSS in Astro
...
+npm i @astrojs/rss
+# yarn
+yarn add @astrojs/rss
+# pnpm
+pnpm i @astrojs/rss
+```
+
+To generate your first RSS feed, create an `rss.xml.js` file under your `src/pages/` directory. `rss.xml` will be the output URL, so feel free to rename this if you prefer.
+
+Now, import the `rss` helper from the `@astrojs/rss` package. We will explore each of the parameters below:
```
Comment (sarah11918) [inline src/pages/en/guides/rss.md:26]:
```suggestion
Now, import the `rss` helper from the `@astrojs/rss` package. We will explore each of the parameters below:
```
```suggestion
Now, import the `rss` helper from the `@astrojs/rss` package and export a function with the following paramaters.
```
Didn't quite seem to be in a consistent voice. Just a suggestion, feel free to make it better/accurate. But, I prefer something like an instruction to export something here.

File: src/pages/en/guides/rss.md [hunk 8]
```diff
@@ -6,11 +6,139 @@ description: An intro to RSS in Astro
...
+    // (optional) inject custom xml
+    customData: `<language>en-us</language>`,
+  });
+```
+
+### Generating `items`
+
+The `items` field accepts either:
+1. [A list of RSS feed objects](#1-list-of-rss-feed-objects), each with a `link`, `title`, `pubDate`, and optional `description` and `customData` fields.
+2. [An `import.meta.glob(...)` result](#2-importmetaglob-result). **Only use this for `.md` files in the `src/pages/` directory!**
```
Comment (sarah11918) [inline src/pages/en/guides/rss.md:50]:
```suggestion
2. [An `import.meta.glob(...)` result](#2-importmetaglob-result). **Only use this for `.md` files within the `src/pages/` directory!**
```

Tagging @delucis and @aFuzzyBear on this very tiny nit. We have tended to use "within the `src/pages/` directory when describing Markdown that lives in any level of subfolders. Our valient (?) attempt to distinguish from "directly in." Might be totally unnecessary, but it's a thing that we've done in general, so just pointing it out.

File: src/pages/en/guides/rss.md [hunk 9]
```diff
@@ -6,11 +6,139 @@ description: An intro to RSS in Astro
...
+
+#### 1. List of RSS feed objects
+
+We recommend this option for `.md` files outside of the `pages` directory. This is common when generating routes [via `getStaticPaths`](/en/reference/api-reference/#getstaticpaths). 
+
+For instance, say our blog posts are stored under a `src/posts/` directory. Each post has a `title`, `pubDate`, and `slug` in its frontmatter, where `slug` corresponds to the output URL on your site. We can generate an RSS feed using [Vite's `import.meta.globEager` helper](https://vitejs.dev/guide/features.html#glob-import) like so:
```
Comment (sarah11918) [inline src/pages/en/guides/rss.md:56]: Maybe instead of the conversational "say our blog posts ..."  I want to see something like:

Example: `.md` posts are located in `src/posts/`

And, I'm not *sure* we need to spell out what the frontmatter values are gonna be? If `slug` is particularly particular, then we can call that out. I know I'm rarely in the "this is overexplaining" camp, but it's a little visually distracting with all the inline code. Would a code example of the yaml front matter be helpful? Could we get away without it entirely, and assume that the corresponding `.md` files are going to have those properties in frontmatter?

Just thinkin' here.

File: src/pages/en/guides/rss.md (hunk 8 above)
Comment (aFuzzyBear) [reply src/pages/en/guides/rss.md]: aye this looks good to me buddy 🙂

File: src/pages/en/guides/rss.md (hunk 9 above)
Comment (bholmesdev) [reply src/pages/en/guides/rss.md]: Hm, I think calling out the frontmatter is pretty necessary since `slug` is worth explaining. I personally find it less distracting than a full-blown `yaml` code block. Idk, I vote keeping that bit 🤷

File: src/pages/en/guides/rss.md (hunk 8 above)
Comment (bholmesdev) [reply src/pages/en/guides/rss.md]: Okay, will keep that consistency in mind!

File: src/pages/en/guides/rss.md (hunk 9 above)
Comment (sarah11918) [reply src/pages/en/guides/rss.md]: Happy to defer on that!

**Comment by sarah11918** [comment]
(A propos of nothing .... does it make you feel any better that I now wish I had just edited the whole file and changed all the import.metas at once?)

File: src/pages/en/guides/rss.md (hunk 9 above)
Comment (bholmesdev) [reply src/pages/en/guides/rss.md]: Sounds good. Made that `.md` post suggestion though!

**Comment by bholmesdev** [comment]
@sarah11918 I mean, that's how I would've done it 😉 Had fun with the commit messages anyways.

This should be all the edits btw. Lmk if it's good to go!

**Comment by sarah11918** [comment]
LGTM!

File: src/pages/en/guides/rss.md [hunk 10]
```diff
@@ -6,11 +6,143 @@ description: An intro to RSS in Astro
...
+    description: 'A humble Astronaut’s guide to the stars',
+    // ex. use your stylesheet from "public/rss/styles.xsl"
+    stylesheet: '/rss/styles.xsl',
+    ...
+  });
+```
+
+## Using `getStaticPaths()`
+
+> **Note:** this approach is more limited compared to `@astrojs/rss`. We mainly include these instructions for legacy purposes.
```
Comment (FredKSchott) [inline src/pages/en/guides/rss.md:141]: I'd rephrase this a bit: 

```suggestion
> **Note:** This method has been deprecated, and will be removed before the official `v1.0.0` release. Please use `@astrojs/rss` instead.
```

That was my understanding of what we were doing here, at least. Happy to chat through more if needed!

File: src/pages/en/guides/rss.md [hunk 11]
```diff
@@ -6,11 +6,143 @@ description: An intro to RSS in Astro
...
+    description: 'A humble Astronaut’s guide to the stars',
+    items: posts.map((post) => ({
+      link: post.frontmatter.slug,
+      title: post.frontmatter.title,
+      pubDate: post.frontmatter.pubDate,
+    }))
+  });
+```
+
+### Adding a stylesheet
```
Comment (FredKSchott) [inline src/pages/en/guides/rss.md:97]: Reading through these docs, I had a thought: It's odd that we have to spend so much time documenting these two different multi-step ways to do something so simple. It's also a bit odd that we generate a static `stylesheet` dynamically like this.

As I user, I realize that what I really want is `stylesheet: '/rss/styles.xsl'`. Instead of vendoring this stylesheet inside the package and making users create two different page routes, why don't we just design for this `public/` workflow only. We could say the following:

--- 

You can style your RSS feed for a more pleasant user experience when viewing the file in your browser.

Use the `rss` function's `stylesheet` option to specify an absolute path to your stylesheet.


```js
rss({
  // ex. use your stylesheet from "public/rss/styles.xsl"
  stylesheet: '/rss/styles.xsl',
  // ...
});
```

If you don't have an RSS stylesheet in mind, we recommend the [Pretty Feed v3](https://github.com/genmon/aboutfeeds/blob/main/tools/pretty-feed-v3.xsl) default stylesheet, which you can download from GitHub and save into your project `public/` directory.

**Review by FredKSchott — COMMENTED** [review]
2 final comments. Would love to chat through in person if you're not on board with either of them!

File: src/pages/en/guides/rss.md (hunk 10 above)
Comment (bholmesdev) [reply src/pages/en/guides/rss.md:141]: I... didn't know if we'd do that! Sounds good if so 👍

File: src/pages/en/guides/rss.md (hunk 11 above)
Comment (bholmesdev) [reply src/pages/en/guides/rss.md:97]: Hm okay, would you want to remove `getStylesheet` + the README entry then?

**Review by FredKSchott — APPROVED** [review]
LGTM!
